Mathlib/Combinatorics/SimpleGraph/Trails.lean

```diff
@@ -156,6 +156,22 @@ theorem IsEulerian.card_odd_degree [Fintype V] [DecidableRel G.Adj] {u v : V} {p
   ext v
   simp
 
+theorem IsEulerian.mem_support_of_Connected {u v : V} {p : G.Walk u v} (hp : p.IsEulerian)
+    (hc : G.Connected) (w : V) : w ∈ p.support := by
+  by_cases h : w = u
+  · simp [h]
+  · have : ∃ e ∈ G.edgeSet, w ∈ e := by
+      obtain ⟨p', _⟩ := hc.preconnected.exists_isPath w u
+      have hw : w ∈ p'.support := by simp
+      obtain ⟨y, _, hwy⟩ := G.adj_of_mem_walk_support p' (not_nil_of_ne h) hw
+      use s(w, y)
+      simp [hwy]
```

Comment:
A matter of style/preference, but if you like you could also make this a one-liner `exact ⟨s(w, y), by simp [hwy]⟩`